### DISCUSSION TITLE
More examples / Git repos recommended for learning Horreum?

### DISCUSSION THREAD

**Discussion by dannepettersson** [Hyperfoil/Horreum#2205] [General]
Hello, 

Im trying to learn Horreum by examples as I understand that the "Getting started" guide on the homepage isn't really up-to-date.  
I have everything up and running via the backend-dev and have loaded the example-data. 

Here the 'quarkus-sb' json files seems to include alot of functionality around the test setup. It would be very very intresting :-) to have test-data also so one could use that as an example to import via the example-configuration.sh script and learn from. 

If you have more example data to learn from or could recommend any open source project that uses Horreum for upload of test-data (python or Java), it would be very appreciated if you could point me to any of these resources. 

I've been looking in @johnaohara github quite alot and found examples there, but if you have more :-) please let me know. 

Thanks in advance, 

Regards Dan

**Comment by lampajr** [comment]
Hi @dannepettersson, thanks a lot for your interest in Horreum 🙏🏻 

>as I understand that the "Getting started" guide on the homepage isn't really up-to-date

yeah, but this only affects how to start Horreum up.. so if you have been already able to start it up via backend-dev that's great!

>Here the 'quarkus-sb' json files seems to include alot of functionality around the test setup. It would be very very intresting :-) to have test-data also so one could use that as an example to import via the example-configuration.sh script and learn from.

I think I can collect some "run" examples that you could upload that would match the `quarkus-sb` test/schema that is already present in https://github.com/Hyperfoil/Horreum/tree/master/infra-legacy/example-data but correct me if you are a looking at something else.

>If you have more example data to learn from or could recommend any open source project that uses Horreum for upload of test-data (python or Java), it would be very appreciated if you could point me to any of these resources.

I can try to look for something like that (@johnaohara do you know if there is some open source project with a full test/schema/run setup?)

In the meantime you can take a look at https://github.com/Hyperfoil/horreum-client-python/blob/main/examples/api_key_usage.py, this is a bare minimum example of test setup using the python client.. unfortunately the data is more or less the same as the one you can find here in `/example-data`.

**Reply by dannepettersson** [reply]
Thanks for your quick reply 👍 

Yes, if you have  some "run" examples with metrics that matches 'quarkus-sb' please upload!

Regards
Dan

**Reply by dannepettersson** [reply]
Hi again @lampajr 

The repos you recommend doesnt have to be a full test/schema/run setup. 

I working in a micro-service system and love working with JVM performance. After finding your solution Im very impressed of the all capabilites built in and I want to make a POC with the diffrent projects at my company. 

But at the moment Im struggeling to see how a  'complete' set-up would look like. 

Thats why Id like too see how other projects use Horreum. Are there no other quarkus microservices project that uses Horreum for example? 

Thanks again for your time. 

/Dan

**Reply by dannepettersson** [reply]
The Jenkins plugin is not an option. Im thinking more of nighly runs that create metrics and upload them to Horreum. But how, and where to start is my main headache at the moment😁

Saw you have a repo for the Java client but not so many examples there unfortunately.

**Reply by lampajr** [reply]
Hey again @dannepettersson, 

There a couple of things here that I'll try to summarize below:

**Setup**
- If you are looking at a complete setup I think the example data we have here in the repo is quite exhaustive as it contains Test and Schemas that are the most important entities in Horreum - you can check more details on those core concepts [here](https://horreum.hyperfoil.io/docs/concepts/core-concepts/)

- Run content usually contains sensitive data as it contains actual performance test results that are not always sharable, that's why it could be hard to see such examples publicly - but this should not prevent you to simply upload some generic JSON that matches the schema you have created (this is usually not enforced, so you can upload any JSON even if it does not match the schema, in that case the labels defined in the Schema will be empty/not filled)

- Once Test and Schemas are setup and you start uploading Runs, the only missing step is setting up Change detection and you can find how to do that at https://horreum.hyperfoil.io/docs/changedetection/

 **Automated upload**
>The Jenkins plugin is not an option. Im thinking more of nighly runs that create metrics and upload them to Horreum

Horreum itself does not provide any capabilities to fetch data periodically and upload it to Horreum, this is left to the User who is interacting with the service. So the "periodic" aspect needs to be implemented on your side, either using and CI/CD system or even using simple cron job or whatever you'd like. Then, when you need to upload Run to Horreum you have several choices:
- Horreum Jenkins plugin (if you are using Jenkins), see https://horreum.hyperfoil.io/docs/integrations/jenkins-plugin/
- Horreum clients (Jaa / Python / Golang), see https://horreum.hyperfoil.io/docs/reference/horreum-clients/
- Plain REST call, e.g., using `curl`, similarly to what https://github.com/Hyperfoil/Horreum/blob/master/infra-legacy/example-configuration.sh is doing

**Reply by dannepettersson** [reply]
I appreciate the info you provided here @lampajr ! and I feel I now have a better idea where to start. 

Also good knowing that the documentation is up-to-date. Its very good and comprehensive. 

Regards
Dan
